File: windows/ClaudeWidget/Models/Account.cs
namespace ClaudeWidget.Models;

public enum AccountKind { Personal, Work }

public static class AccountKindExtensions
{
    public static string DisplayName(this AccountKind kind) => kind switch
    {
        AccountKind.Personal => "Personal",
        AccountKind.Work => "Work",
        _ => "Unknown"
    };

    /// <summary>Segoe Fluent Icons glyph code (for use with FontFamily="Segoe Fluent Icons").</summary>
    public static string Glyph(this AccountKind kind) => kind switch
    {
        AccountKind.Personal => "", // Contact
        AccountKind.Work     => "", // Work
        _ => ""
    };

    /// <summary>Stable string for storage keys.</summary>
    public static string Key(this AccountKind kind) => kind switch
    {
        AccountKind.Personal => "personal",
        AccountKind.Work     => "work",
        _ => "unknown"
    };
}

/// <summary>One usage window (5-hour, 7-day, etc.) as claude.ai reports it.</summary>
public sealed record UsageWindow(string Label, double Utilization, DateTimeOffset? ResetsAt);

/// <summary>Credits-based extra usage (the paid add-on visible in claude.ai → Settings → Usage when enabled).</summary>
public sealed record ExtraUsage(
    bool IsEnabled,
    double MonthlyLimit,   // dollars (already normalized from cents)
    double UsedCredits,    // dollars (already normalized from cents)
    double Utilization,    // 0..1
    string Currency);

public sealed record AccountUsage(
    IReadOnlyList<UsageWindow> Windows,
    ExtraUsage? Extra,
    string PlanLabel,
    DateTimeOffset LastUpdated)
{
    /// <summary>Most-utilized active window — the one the user most likely cares about.</summary>
    public UsageWindow? PrimaryWindow =>
        Windows.Count == 0 ? null : Windows.OrderByDescending(w => w.Utilization).First();

    /// <summary>
    /// Highest utilization 0..1 across all active limits on this account
    /// (windows + extra_usage). Drives the menu-bar percentage badge.
    /// </summary>
    public double PeakUtilization
    {
        get
        {
            double peak = 0;
            foreach (var w in Windows) peak = Math.Max(peak, w.Utilization);
            if (Extra is { IsEnabled: true }) peak = Math.Max(peak, Extra.Utilization);
            return Math.Min(1, peak);
        }
    }
}

public sealed record Organization(string Uuid, string Name);

public sealed class Account
{
    public required AccountKind Kind { get; init; }
    public string Label { get; set; } = "";
    public bool IsConfigured { get; set; } = true;
    public AccountUsage? Usage { get; set; }
    public List<Organization> AvailableOrgs { get; set; } = new();
    public string? SelectedOrgUuid { get; set; }
}

File: Sources/Account.swift
import Foundation

enum AccountKind: String, CaseIterable, Identifiable, Codable {
    case personal
    case work

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .personal: return "Personal"
        case .work:     return "Work"
        }
    }

    var icon: String {
        switch self {
        case .personal: return "person.crop.circle"
        case .work:     return "briefcase"
        }
    }
}

/// One usage window (5-hour, 7-day, 7-day-opus, 7-day-sonnet, etc.) as
/// claude.ai reports it: a utilization percentage 0…1 and an optional reset
/// timestamp. `nil` whole-struct means the window is inactive / not applicable
/// (e.g. a Pro user has no 7-day-opus window).
struct UsageWindow: Equatable {
    let label: String       // "5-hour", "7-day", "Opus 7-day", "Sonnet 7-day"
    let utilization: Double // 0…1 (fraction; claude.ai sends 0…100, we normalize)
    let resetsAt: Date?
}

/// Credits-based extra usage (the paid add-on shown in the desktop app when
/// you've enabled "extra usage" pay-as-you-go on top of Pro/Max).
struct ExtraUsage: Equatable {
    let isEnabled: Bool
    let monthlyLimit: Double  // dollars
    let usedCredits: Double   // dollars
    let utilization: Double   // 0…1
    let currency: String
}

struct AccountUsage: Equatable {
    var windows: [UsageWindow]
    var extra: ExtraUsage?
    var planLabel: String
    var lastUpdated: Date

    /// Most-utilized active window — the one the user most likely cares about.
    var primaryWindow: UsageWindow? {
        windows.max { $0.utilization < $1.utilization }
    }

    /// 0…1 remaining fraction across the most-utilized window (used by the
    /// menu bar percentage badge). Defaults to 1 (full) if no windows are active.
    var remainingPercent: Double {
        guard let w = primaryWindow else { return 1.0 }
        return max(0, min(1, 1.0 - w.utilization))
    }
}

struct Account: Identifiable, Equatable {
    let kind: AccountKind
    var label: String
    var isConfigured: Bool
    var usage: AccountUsage?

    var id: String { kind.id }
}
